webapp/src/modules/tile/sagas.js

```diff
@@ -15,12 +15,18 @@ export function* tileSaga() {
 
 function* handleTilesRequest(action) {
   try {
-    const nw = buildCoordinate(action.nw.x, action.nw.y)
-    const se = buildCoordinate(action.se.x, action.se.y)
-    const address = yield select(getAddress)
-    const tiles = yield call(() => api.fetchTiles(nw, se, address))
-
+    const tiles = yield call(() => api.fetchTiles())
     yield put(fetchTilesSuccess(tiles))
+
+    let address = yield select(getAddress)
+
+    if (!address) {
+      const action = yield take(CONNECT_WALLET_SUCCESS)
```

Comment:
What do you think of moving this `take` to another saga which fetches the address tiles and updates them? We could avoid a race condition